### PR TITLE
build(eslint-plugin-odyssey): lint for filename and dirname in theme properties

### PR DESCRIPTION
**Pull request by jeffbelser-okta**

### CODE REVIEW

**Comment by jeffbelser-okta** [comment]
VRT failure is expected here until #1284 is merged

File: packages/odyssey-react/src/components/Text/Text.theme.ts [hunk 1]
```diff
@@ -14,11 +14,13 @@ import type { ThemeReducer } from "@okta/odyssey-react-theme";
 
 export const theme: ThemeReducer = (theme) => ({
   // Font
+  /* eslint-disable @okta/odyssey/no-invalid-theme-properties */
   ColorTextBody: theme.ColorTextBody,
```
Comment (edburyenegren-okta) [inline packages/odyssey-react/src/components/Text/Text.theme.ts:18]: Which rules are these violating?

**Review by edburyenegren-okta — APPROVED** [review]
Do we have any docs around expected format? I can add them to the Tokens naming sheet if not.

File: packages/odyssey-react/src/components/Text/Text.theme.ts (hunk 1 above)
Comment (jeffbelser-okta) [reply packages/odyssey-react/src/components/Text/Text.theme.ts:18]: @edburyenegren-okta the `filename` rule since they contain `Text` 🙄

**Comment by jeffbelser-okta** [comment]
@edburyenegren-okta only change here is to avoid including the name of the Component itself. That leads to duplication in the run time API we want to avoid:

```
theme={{
  [Button.theme]: { ButtonPaddingBlock: "..." }
}}
```

**Pull request merged**